addons/account/wizard/account_move_reversal_view.xml

```diff
@@ -12,7 +12,7 @@
                     <field name="move_type" invisible="1"/>
                     <field name="available_journal_ids" invisible="1"/>
                     <group>
-                         <group attrs="{'invisible': [('move_type', 'not in', ('out_invoice', 'in_invoice'))]}">
+                         <group attrs="{'invisible': ['|', ('move_type', 'not in', ('out_invoice', 'in_invoice')), ('residual', '=', 0)]}">
```

Comment:
I don't understand this one. Why are we looking at the residual amount? The fact that it's been paid doesn't mean that we can't do a credit note or change the refund method.

Reply:
It's in the PR/commit description.

1) (doesn't concern this part of the code) 
When amount_total is 0,we can't click on the button to get into the wizard.
When residual is 0, I need to be able to click on the button in order to get the wizard.

2) when residual is 0, I can only create a reverse for 100% of the move.
In the current behavior, the radio button are disabled (read only).
As I can't chose anything here, I want to hide it.
Thus,when residual is 0,I hide the group of the radio buttons.

Reply:
> when residual is 0, I can only create a reverse for 100% of the move.

What? Why?
* I invoiced the client for 3 items
* He paid for it
* He reports after that one of the items is broken and that he wants to be refunded

Now I'd like to refund 1/3 of the invoice, not the full invoice. Even though the residual is 0. Am I missing something?

Reply:
While writing this and checking functionally the diff between 14.0 and 15.0, I realized that I'm fixing a symptom of another problem.
--> In 14.0, when the invoice is already reconciled, if we make a refund, the behavior is exactly the same as now BUT as the radio buttons are disabled and the default option selected, only the default option appears and the radio button doesn't appear as one.
Maybe the real issue is that `readonly` attribute from next line `<field name="refund_method" widget="radio" attrs="{'readonly': [('residual', '=', 0)]}"/>` doesn't behave the same way anymore.

explanation I was writing before that:
---------------------------
I'm probably explaining it wrong.

The current behavior:
- you create an invoice with 3 product a, b, c
- you click reverse
- you have to choose between:
    - partial refund
    - full refund
    - full refund and new draft invoice 

If you choose:
- partial refund (the default), it creates a reverse move in draft for 100% of those amount and **doesn't reconcile** the initial invoice and the reverse.
- full refund, it creates a reverse move in draft for 100% of those amount and **reconcile** the initial invoice and the reverse. (which label the invoice as "reversed" and the reverse as "paid")
- full refund and new draft invoice, same than previous but open a draft invoice

If the residual is 0, choosing full refund would break the full reconciliation. Thus we wan't to prevent that. Which is probably why the choice has been impossible.

Reply:
https://github.com/odoo/odoo/commit/288b24cbdf54ac0dbee7e012f074fac9a0c68238

Reply:
I think it's linked to this commit: https://github.com/odoo/odoo/commit/288b24cbdf54ac0dbee7e012f074fac9a0c68238

Reply:
Regarding your first explanation, it should then break the full reconciliation and add a partial reconcile instead of not allowing the option

Reply:
> Regarding your first explanation, it should then break the full reconciliation and add a partial reconcile instead of not allowing the option

Yes and No, It's a matter of opinion.
You can consider that the invoice go with the payment and the credit note with the refund payment, separately.
You can also consider that the 4 of them are related and belong to each other together.

Depending on the situation, one or the other is preferable.

Reply:
This is really "meh". We should rework the refund method options anyways (in master)

Reply:
I think we should give the choice to the user in the wizard with a default one selected.